src/fsharp/CheckFormatStrings.fs

```diff
@@ -88,65 +98,65 @@ let parseFormatStringInternal (m:range) (g: TcGlobals) (context: FormatStringChe
                     flags(i+1)
                 | '+' -> 
-                    if info.numPrefixIfPos <> None then failwithf "%s" <| FSComp.SR.forPrefixFlagSpacePlusSetTwice()
+                    if info.numPrefixIfPos <> None then raise (Failure (FSComp.SR.forPrefixFlagSpacePlusSetTwice()))
                     info.numPrefixIfPos <- Some '+'
                     flags(i+1)
                 | '0' -> 
-                    if info.addZeros then failwithf "%s" <| FSComp.SR.forFlagSetTwice("0")
+                    if info.addZeros then raise (Failure (FSComp.SR.forFlagSetTwice("0")))
                     info.addZeros <- true
                     flags(i+1)
                 | ' ' -> 
-                    if info.numPrefixIfPos <> None then failwithf "%s" <| FSComp.SR.forPrefixFlagSpacePlusSetTwice()
+                    if info.numPrefixIfPos <> None then raise (Failure (FSComp.SR.forPrefixFlagSpacePlusSetTwice()))
                     info.numPrefixIfPos <- Some ' '
                     flags(i+1)
-                | '#' -> failwithf "%s" <| FSComp.SR.forHashSpecifierIsInvalid() 
+                | '#' -> raise (Failure (FSComp.SR.forHashSpecifierIsInvalid() ))
                 | _ -> i
 
               let rec digitsPrecision i = 
-                if i >= len then failwithf "%s" <| FSComp.SR.forBadPrecision()
+                if i >= len then raise (Failure (FSComp.SR.forBadPrecision()))
                 match fmt.[i] with
                 | c when System.Char.IsDigit c -> digitsPrecision (i+1)
                 | _ -> i 
 
               let precision i = 
-                if i >= len then failwithf "%s" <| FSComp.SR.forBadWidth()
+                if i >= len then raise (Failure (FSComp.SR.forBadWidth()))
                 match fmt.[i] with
                 | c when System.Char.IsDigit c -> info.precision <- true; false,digitsPrecision (i+1)
                 | '*' -> info.precision <- true; true,(i+1)
-                | _ -> failwithf "%s" <| FSComp.SR.forPrecisionMissingAfterDot()
+                | _ -> raise (Failure (FSComp.SR.forPrecisionMissingAfterDot()))
 
               let optionalDotAndPrecision i = 
-                if i >= len then failwithf "%s" <| FSComp.SR.forBadPrecision()
+                if i >= len then raise (Failure (FSComp.SR.forBadPrecision()))
                 match fmt.[i] with
                 | '.' -> precision (i+1)
                 | _ -> false,i
 
-              let rec digitsWidthAndPrecision i = 
-                if i >= len then failwithf "%s" <| FSComp.SR.forBadPrecision()
+              let rec digitsWidthAndPrecision n i = 
+                if i >= len then raise (Failure (FSComp.SR.forBadPrecision()))
                 match fmt.[i] with
-                | c when System.Char.IsDigit c -> digitsWidthAndPrecision (i+1)
-                | _ -> optionalDotAndPrecision i
+                | c when System.Char.IsDigit c -> digitsWidthAndPrecision (n*10 + int c - int '0') (i+1)
```

Comment:
What is the significance of `n*10 + int c - int '0'`?

Reply:
With interpolated strings we need to extract the width field from the format specifier to push it into the .NET format string should we need to use it, see https://github.com/dotnet/fsharp/pull/8907/files#diff-1e32fb5e1febaac321f005ea032fc833R231

The `int c - int '0'` is computing the numeric value of a digit character, the `n*10` is multiplying it up character by character